src/devices/niko.ts

```diff
@@ -119,50 +166,100 @@ const local = {
+                } else {
+                    await entity.write(
+                        'manuSpecificNikoState',
+                        // @ts-expect-error ignore
+                        {switchActionReporting: actionReportingMap[value]},
+                    );
+                    return {state: {action_reporting: value}};
+                }
+            },
+            convertGet: async (entity, key, meta) => {
+                await entity.read('manuSpecificNikoState', ['switchActionReporting']);
             },
         } satisfies Tz.Converter,
         switch_led_enable: {
             key: ['led_enable'],
             convertSet: async (entity, key, value, meta) => {
-                await entity.write('manuSpecificNiko1', {outletLedState: value ? 1 : 0});
-                await entity.read('manuSpecificNiko1', ['outletLedColor']);
+                await entity.write('manuSpecificNikoConfig', {outletLedState: value ? 1 : 0});
+                await entity.read('manuSpecificNikoConfig', ['outletLedColor']);
                 return {state: {led_enable: value ? true : false}};
             },
             convertGet: async (entity, key, meta) => {
-                await entity.read('manuSpecificNiko1', ['outletLedState']);
+                await entity.read('manuSpecificNikoConfig', ['outletLedState']);
             },
         } satisfies Tz.Converter,
         switch_led_state: {
             key: ['led_state'],
             convertSet: async (entity, key, value, meta) => {
-                utils.assertString(value, key);
-                await entity.write('manuSpecificNiko1', {outletLedColor: value.toLowerCase() === 'off' ? 0 : 255});
-                return {state: {led_state: value.toLowerCase() === 'off' ? 'OFF' : 'ON'}};
+                const ledStateMap: KeyValue = {OFF: 0x00, ON: 0x0000ff, Blue: 0x00ff00, Red: 0xff0000, Purple: 0xffffff};
+                // @ts-expect-error ignore
+                if (ledStateMap[value] === undefined) {
+                    throw new Error(`led_state was called with an invalid value (${value})`);
+                } else {
+                    await entity.write(
+                        'manuSpecificNikoConfig',
+                        // @ts-expect-error ignore
+                        {outletLedColor: ledStateMap[value]},
+                    );
+                    return {state: {led_state: value}};
+                }
             },
             convertGet: async (entity, key, meta) => {
-                await entity.read('manuSpecificNiko1', ['outletLedColor']);
+                await entity.read('manuSpecificNikoConfig', ['outletLedColor']);
+            },
+        } satisfies Tz.Converter,
+        switch_led_sync_mode: {
+            key: ['led_sync_mode'],
+            convertSet: async (entity, key, value, meta) => {
+                // This could be set endpoint specific, where the first 4 bits are used for the left button and the last 4 bits for the right button
```

Comment:
Any pointers on how to make this endpoint specific are welcome. I need the value for both endpoints to write the correct value to the device..

Reply:
Use the `meta.state` (this is identical to the state that is published to MQTT)

Reply:
I managed to get this working for multiple endpoints. Thanks!